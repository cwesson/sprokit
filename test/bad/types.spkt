
type A {
	var v : int32;
}

func test() : int32 {
	var a : int8 = 14569;
	var b : int16 = 4562156;
	var c : A = 12;
	var d : int32 = 159 + true;
	if 13 - 13 {
		return 0;
	}
	return false;
}
