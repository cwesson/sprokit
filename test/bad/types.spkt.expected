Error on line 7:
  Cannot assign int16 to int8
Error on line 8:
  Cannot assign int32 to int16
Error on line 9:
  Cannot assign int8 to A
Error on line 10:
  Cannot add int8 and bool
Error on line 10:
  Cannot assign unknown type to int32
Error on line 11:
  Cannot convert int8 to bool
Error on line 14:
  Cannot return bool, expecting int32
